src/realm/sync/noinst/sync_metadata_schema.hpp

```diff
@@ -120,19 +120,35 @@ struct SyncMetadataTable {
 void create_sync_metadata_schema(const TransactionRef& tr, std::vector<SyncMetadataTable>* tables);
 void load_sync_metadata_schema(const TransactionRef& tr, std::vector<SyncMetadataTable>* tables);
 
-// SyncMetadataSchemas manages the schema version numbers for different groups of internal tables used
-// within sync.
-class SyncMetadataSchemaVersions {
+class SyncMetadataSchemaVersionsReader {
 public:
-    explicit SyncMetadataSchemaVersions(const TransactionRef& ref);
+    explicit SyncMetadataSchemaVersionsReader(const TransactionRef& ref);
 
     util::Optional<int64_t> get_version_for(const TransactionRef& tr, std::string_view schema_group_name);
-    void set_version_for(const TransactionRef& tr, std::string_view schema_group_name, int64_t version);
 
-private:
+    // returns false if the metadata schema versions table is not available, in cases where:
+    // * the metadata schema version table was not created during construction (i.e. read-only flag was set)
+    // * the metadata schema version table could not be read
+    // * the legacy metadata schema version data still exists and has not been converted
+    bool is_initialized() const
```

Comment:
The comment is not accurate anymore. Do we need this method? Since `get_version_for` returns an optional, there is no problem if the metadata is not available.

Reply:
Yeah, I think you're right - I was using the `is_initialized()` to indicate whether or not the data could be read, but, since the optional value is returned by `get_version_for()`, it doesn't matter whether the table existed or that the value hadn't been written yet. Either way, the version is not available and the data will need to be written at some point before it can be used.

I'll remove it.